jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpParser.java

```diff
@@ -454,15 +460,21 @@ private HttpTokens.Token next(ByteBuffer buffer)
                 if (_cr)
                     throw new BadMessageException("Bad EOL");
 
-                _cr = true;
                 if (buffer.hasRemaining())
                 {
                     // Don't count the CRs and LFs of the chunked encoding.
                     if (_maxHeaderBytes > 0 && (_state == State.HEADER || _state == State.TRAILER))
                         _headerBytes++;
-                    return next(buffer);
+                    ch = buffer.get();
+                    t = HttpTokens.TOKENS[0xff & ch];
+                    return switch (t.getType())
+                    {
+                        case CNTL -> throw new IllegalCharacterException(_state, t, buffer);
+                        case LF -> t;
+                        default -> throw new BadMessageException("Bad EOL");
+                    };
```

Comment:
The recursive call was doing more options that are not appropriate and hence had to be guarded by setting, then testing and maybe clearing _cr.   This is much simpler, if the next charater is availbel and it is LF, then we have a LF otherwise anything after a CR is an exception of one sort or another.

Reply:
To be more specific, every single CRLF in a header was being handled by:
 + switch on CR
 + setting _cr true
 + recursing
 + switch on LF
 + clearing _cr.

Now we just:
 + switch on CR
 + switch on LF